tt_eager/tt_dnn/op_library/move/move_op.hpp

```diff
@@ -102,7 +109,6 @@ inline Tensor move(Tensor& input_tensor, std::optional<MemoryConfig>& mem_config
     }
 
     auto output = operation::run(Move{output_mem_config, move_op_parallelization_strategy}, {input_tensor, output_tensor}).at(0);
-    input_tensor.deallocate();
```

Comment:
we call DeallocateBuffer(input_tensor.buffer()) before creating output tensor and then we were deallocating the input a second time. 

In the test we don't actually end up reallocating because the input tensor to the move op is already at the highest l1 address that it could be allocated in. 

Sequence of steps that caused the issue:
```
allocate tensor 0 at address A
allocate tensor 1 at address B 
move(tensor 0)
   - deallocate tensor 0 from address A via DeallocateBuffer(*input_tensor.buffer()) line 60
   - create new tensor --> it gets allocated at address A again 
   - move data 
   - input_tensor.deallocate() --> marks address A as free even though the output tensor is now at A
move(tensor 1)
   - deallocate tensor 1 from address B via  DeallocateBuffer(*input_tensor.buffer()) line 60
   - create new tensor --> gets allocated at address A because this was marked as free 
   - move data from tensor 1 into address A
   -  input_tensor.deallocate() --> marks address B as free again

error because we lost tensor 0 data 
```